entitlement management: add additionalAccess query API for separation of duties analysis (#16495)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accesspackageassignment-additionalaccess.md

---
title: "accessPackageAssignment: additionalAccess"
description: "Retrieve a list of accesspackageassignment objects indicating potential separation of duties conflicts or additional access."
ms.localizationpriority: medium
author: "markwahl-msft"
ms.prod: "governance"
doc_type: "apiPageType"
---

# accessPackageAssignment: additionalAccess
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> In [Azure AD Entitlement Management](../resources/entitlementmanagement-overview.md), retrieve a collection of [accessPackageAssignment](../resources/accesspackageassignment.md) objects that indicate additional access for a target user.  This can be used to prepare to configure the incompatible access packages for a specific access package.

Reviewer: Same comment+Q as in the metadata - Looking at the syntax and the rest of the query patterns below, sounds like the API checks both through the specified IDs. Please confirm if this suggestion is technically correct.
```suggestion
In [Azure AD Entitlement Management](../resources/entitlementmanagement-overview.md), retrieve a collection of [accessPackageAssignment](../resources/accesspackageassignment.md) objects that indicate access to a specified access package and an incompatible access package. This can be used to prepare to configure the incompatible access packages for a specific access package.
```

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|EntitlementManagement.Read.All, EntitlementManagement.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported.|
|Application|Not supported.|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http

> GET /identityGovernance/entitlementManagement/accessPackageAssignments/additionalAccess(accessPackageId=`parameterValue`,incompatibleAccessPackageId=`parameterValue`)

Reviewer: ```suggestion
GET /identityGovernance/entitlementManagement/accessPackageAssignments/additionalAccess(accessPackageId='parameterValue',incompatibleAccessPackageId='parameterValue')
```

```

## Function parameters

> The following table shows the parameters that can be used with this function.

Reviewer: The API.md indicates that the function parameters are mandatory.
```suggestion
The following table shows the parameters that must be supplied with this function.
```

|Parameter|Type|Description|
|:---|:---|:---|
| accessPackageId | String |  Indicates the ID of an access package for which the caller would like to retrieve the assignments. |

> | incompatibleAccessPackageId | String | Returns the assignments for users who also have assignments to another, incompatible, access package. |

Author: updated to mark both as required.

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|

## Request body
Do not supply a request body for this method.

## Response

> If successful, this method returns a `200 OK` response code and an [accessPackageAssignment](../resources/accesspackageassignment.md) collection in the response body.

Reviewer: ```suggestion
If successful, this method returns a `200 OK` response code and a collection of [accessPackageAssignment](../resources/accesspackageassignment.md) objects in the response body.
```

> When a result set spans multiple pages, Microsoft Graph returns that page with an `@odata.nextLink` property in the response that contains a URL to the next page of results. If that property is present, continue making additional requests with the `@odata.nextLink` URL in each response, until all the results are returned. For more information, see [paging Microsoft Graph data in your app](/graph/paging).

Reviewer: Sounds like the API also supports optional query parameters, probably `$top` and `$select` to start us off. Please feel free to add an "Optional query parameters" H2 section after the "Function parameters" section and before the "Request headers" section.

## Examples

The following example gets the access package assignments for users who have assignments to both access packages.

### Request

<!-- {
  "blockType": "request",
  "name": "accesspackageassignment_additionalaccess"
}
-->
``` http
GET https://graph.microsoft.com/beta/identityGovernance/entitlementManagement/accessPackageAssignments/additionalAccess(accessPackageId='4c6bc7cc-c1f4-450e-8e90-2984244ef0fd',incompatibleAccessPackageId='2fda5e0a-459e-eb11-8d2a-9c7bef3841d6')
```

### Response
> **Note:** The response object shown here might be shortened for readability.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "Collection(microsoft.graph.accessPackageAssignment)"
}
-->
``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "value": [
        {
            "@odata.type": "#microsoft.graph.accessPackageAssignment"
        }
  ]

> }

Reviewer: Please share a real and "whole" response object. This sample indicates there aren't any instances of the assignments - so the response object should probably look different anyway.

Author: added
